pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -339,6 +364,15 @@ func (d *DeploymentController) apply(controller string, yml string) error {
 	return nil
 }
 
+func (d *DeploymentController) HandleTagChange(newTags []string) {
+	d.myTags = sets.New(newTags...)
+	for _, gw := range d.gateways.List(metav1.NamespaceAll, klabels.Everything()) {
+		if string(gw.Spec.GatewayClassName) == gw.GetName() {
```

Comment:
GatewayClassName will never == gw.GetName (unless its a coincidence)

Reply:
I fixed that once.  It must have crept back into the PR when I switched dev environments...